lib/index.js

```diff
@@ -51,30 +52,45 @@ cliCompleter.pluginId = constants.sourceId
  * Plugin to validate that Markdown links and images reference existing local
  * files and headings.
  *
- * @type {import('unified').Plugin<[Options?, FileSet?]|[Options?]|void[], Root>}
+ * @typedef {import('unified-lint-rule').Label|import('unified-lint-rule').Severity} Severity
+ * @type {import('unified').Plugin<[(Options|[Severity, Options?])?, FileSet?], Root>}
+ * https://github.com/microsoft/TypeScript/pull/48132
+ * @param [rawOptions]
+ * @param [fileSet]
  */
-export default function remarkValidateLinks(options, fileSet) {
+export default function remarkValidateLinks(rawOptions, fileSet) {
   // Attach a `completer`.
   if (fileSet) {
     fileSet.use(cliCompleter)
   }
 
   // Find references and landmarks.
-  return (tree, file, next) => {
-    find.run(
-      {tree, file, fileSet, options: {...options}},
-      /** @type {Callback} */
-      (error) => {
-        if (error) {
-          next(error)
-        } else if (fileSet) {
-          next()
-        } else {
-          checkAll([file], next)
+  return lintRule(
+    {
+      origin: 'validate-links',
```

Comment:
This most likely prevents `ruleId`s from being set. We currently use different `ruleId`s, and this overwrites them: https://github.com/remarkjs/remark-validate-links/blob/78871b024467c3eeff6efa8512e928438ed98b14/lib/check/validate.js#L73-L88.

Reply:
It’s likely not correct to use use `lintRule`. That’s meant for **one rule** (one reason for emitting things). This package supports **different reasons**.
Perhaps `coerce` should be used, and exposed from `unified-lint-rule`?